File: Medium/reverse_integer.go
package main

import (
	"fmt"
	"strconv"
)

func reverse(x int) int {
	strX := strconv.Itoa(x)
	isNegative := string(strX[0]) == "-"

	if isNegative {
		strX = "-" + reverseString(strX[1:len(strX)])
	} else {
		strX = reverseString(strX)
	}

	convToInteger, err := strconv.Atoi(strX)
	if err != nil {
		return 0
	}

	if convToInteger < -(1<<31) || convToInteger > (1<<31)-1 {
		return 0
	}

	return convToInteger
}

func run_reverse() {
	fmt.Println(reverse(123))        //321
	fmt.Println(reverse(-123))       //-321
	fmt.Println(reverse(1534236469)) //0
}

File: Medium/reverse_integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
    let strX = x.toString();
    const isNegative = strX[0] == "-";

    if(isNegative) {
        strX = "-" + strX.substring(1, strX.length).split("").reverse().join("");
    } else {
        strX = strX.split("").reverse().join("");
    }

    let convToInteger = parseInt(strX);

    if (convToInteger < -(2 ** 31) || convToInteger > (2 ** 31) - 1) {
        return 0;
    }

    return convToInteger;
};

console.log(reverse(123)); //321
console.log(reverse(-123)); //-321
console.log(reverse(1534236469)); //0
